### ISSUE TITLE
Data augmentation for validation set during training ? 

### ISSUE THREAD

**Issue by ThiwankiDias** [ultralytics/yolov5#9554]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi sir,

I m very beginner to YOLOv5 and object detection. I researched about mosaic and albumentation augmentations. I know we can use them for augmentation during training for train dataset in yolov5. but I have a problem is there a way to do data augmentation during training for validation set. should I change hyps or augmentation.py file for that?  if yes please say how to do it. 

Thank you so much for any help

**Comment by glenn-jocher** [comment]
@ThiwankiDias no, augmentations is typically never performed on a validation set, only on training data.

**Comment by karantai** [comment]
@glenn-jocher Yet, if we do not augment validation set during training, validation loss will be lower than training loss, sometimes.
We can augment our data with val.py, but is there an option of augment validation set during training in order for the plots to be correct, that is, the validation loss to be higher than training loss.
Maybe I miss something, that's why I am asking! 
Thanks in advance.

**Comment by glenn-jocher** [comment]
@karantai yes it's expected that train loss is higher than val loss when augmentation is used.

**Comment by stephansturges** [comment]
It would be great to have the option to augment val data in a simple way using the same augmentations as the training set... there are use cases where you are trying to simulate something using augmentations on the val data, and don't have "real world" data for validation that corresponds to what you will be using in a real situation. In these cases it would be great to be able to have an augmentation stack apply to both training and validation in the same way (or be able to define specific set of augmentations for validation).

**Comment by glenn-jocher** [comment]
@stephansturges absolutely, I understand the use case you've mentioned. While YOLOv5 doesn't currently support data augmentation for the validation set during training, I appreciate your feedback and will share it with the team for consideration. Thank you for sharing your perspective!

**Comment by knotgrass** [comment]
@glenn-jocher do you have a plan to support data augmentation for val dataset. I'm eagering to know your view.
From my point of view, this feature is very helpful if we can configuration separately train & val augmentation.
